### PR TITLE
Docker container

### PR DESCRIPTION
**Pull request by KeithLev**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.7-slim-buster AS base
+EXPOSE 5000
+WORKDIR /todo-app
+RUN apt-get update
+RUN apt-get install -y curl
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+ENV PATH="${PATH}:/root/.poetry/bin"
+COPY ./poetry.toml ./pyproject.toml ./
+RUN poetry install
+
```
Comment (hugh-emerson) [inline Dockerfile:10]: Your image currently ONLY works when you run it with a bind mount. That's not what we want. At the very least, the production image needs to run without a bind mount - the production server should be able to download and run the image without worrying cloning this repository and mounting a folder.

So add a COPY command that copies your todo_app folder into the image, at least for the production stage.

File: README.md [hunk 2]
```diff
@@ -68,3 +68,24 @@ You should see 4 tests pass
...
+Run the containers using the following commands:
+Development:
+docker run -p 5000:5000 --env-file .env --mount type=bind,source="$(pwd)"/todo_app,target=/todo-app/todo_app todo-app:dev
+Production:
+docker run -p 5000:5000 --env-file .env --mount type=bind,source="$(pwd)"/todo_app,target=/todo-app/todo_app todo-app:prod
+
+## Run Docker Compose
+Ensure .env file is added to base dicetory
+
+Run "docker compose" from the comand line
```
Comment (hugh-emerson) [inline README.md]: You mean `docker compose up`?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.7-slim-buster AS base
+EXPOSE 5000
+WORKDIR /todo-app
+RUN apt-get update
+RUN apt-get install -y curl
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+ENV PATH="${PATH}:/root/.poetry/bin"
+COPY ./poetry.toml ./pyproject.toml ./
```
Comment (hugh-emerson) [inline Dockerfile]: You ought to copy in the poetry.lock file to make sure you use the exact same versions of all your dependencies

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    volumes:
+      - ./todo_app:/todo-app/todo_app
+    env_file:
+      - ./.env
+    
+
+  prod:
+    build:
+      context: .
+      target: development
```
Comment (hugh-emerson) [inline docker-compose.yaml]: The service names dev/prod are the wrong way around compared to what they're actually targeting (production/development respectively)

**Review by hugh-emerson — CHANGES_REQUESTED** [review]
Good work! Unfortunately, you're missing a COPY command that you need, but the fix should be straightforward. 

My other comments are just small suggestions

File: docker-compose.yaml (hunk 4 above)
Comment (KeithLev) [reply docker-compose.yaml]: *facepalm*

**Pull request merged**